File: 0735-asteroid-collision/0735-asteroid-collision.py
class Solution(object):
    def asteroidCollision(self, asteroids):
        """
        :type asteroids: List[int]
        :rtype: List[int]
        """
        stack=[]
        res=[]
        for i in asteroids:
            if not stack and i<0:
                res.append(i)
            elif i>0:
                stack.append(i)
            elif stack and i<0:
                c,e=0,0
                while stack:
                    if abs(stack[-1])>abs(i):
                        break
                    elif abs(stack[-1])<abs(i):
                        c+=1
                        stack.pop()
                    else:
                        stack.pop()
                        e+=1
                        break
                if c!=0 and not stack and e==0:
                    res.append(i)
        if stack:
            for i in stack:
                res.append(i)
        return res

File: 0735-asteroid-collision/0735-asteroid-collision.java
import java.util.Stack;

class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> stack = new Stack<>();
        
        for (int a : asteroids) {
            // Process the current asteroid
            while (!stack.isEmpty() && stack.peek() > 0 && a < 0) {
                int top = stack.peek();
                if (top < -a) {
                    stack.pop(); // Remove the top asteroid as it is destroyed
                } else if (top == -a) {
                    stack.pop(); // Both asteroids are destroyed
                    a = 0; // Set to 0 as current asteroid is also destroyed
                } else {
                    a = 0; // Current asteroid is destroyed
                }
            }
            if (a != 0) {
                stack.push(a); // Add remaining asteroid to the stack
            }
        }
        int ansArr[] = new int [stack.size()];
        for(int i=stack.size()-1;i>=0;i--){
            ansArr[i] = stack.pop();
        }
        return ansArr;
    }
}
